src/exports.cc

```diff
@@ -236,10 +236,10 @@ Word call_foreign_function(void *raw_context, Word function_name, Word function_
     result_size = s;
     return result;
   });
-  if (!context->wasmVm()->setWord(results, Word(address))) {
+  if (address && !context->wasmVm()->setWord(results, Word(address))) {
```

Comment:
Erm, the other way around, this should test for `(results && !context...)`.

Reply:
I need to patch this up a bit. The problem is that the if the user doesn't provide an address but the foreign function allocates memory then that memory is allocated in the VM.  Since we removed _free, I have no way of freeing it.  I will create a temporary location and then toss the result.  OK, PTAL